### PR TITLE
Pallet-pools: replace Vec storage by BoundedVec

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

Replace `Vec` storage with `BoundedVec` in `pallet-pools` to be prepared for [Weights V2](https://github.com/paritytech/substrate/pull/10918)

Related to: #897

## Changes and Descriptions

Modify `Vec` types found in the `pallet-pools` storages.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue). In this case, a future issue.

# How Has This Been Tested?

```sh
cargo test -p pallet-pools
cargo test --workspace --release --features test-benchmarks,try-runtime
cargo run --features runtime-benchmarks benchmark pallet --pallet="pallet-pools" --chain="altair-dev" --extrinsic="*"
```

# Checklist:
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] I rebased on the latest `parachain` branch

### CODE REVIEW

File: pallets/pools/src/lib.rs
Comment (lemunozm) [inline pallets/pools/src/lib.rs:440]: I've used for the bound the `MaxTranches` which seems to be the right value. Can Anyone involved in `pallet-pools` confirm this?

**Review by NunoAlexandre — DISMISSED** [review]
Looks good to me but it should be approved by a pallet-pools pool owner ☑️

File: pallets/pools/src/lib.rs
Comment (mustermeiszer) [reply pallets/pools/src/lib.rs:440]: This is correct.

**Review by mustermeiszer — COMMENTED** [review]
This looks fine. 

The reason why I am not approving is that I am not sure of the bounds of vectors are checked when being deserialized from rpc. 

Can anybody confirm that an extrinsic with an `BoundedVec<T, S>` will fail to be called when the input for the vector exceeds `S`?

File: pallets/pools/src/lib.rs
Comment (branan) [inline pallets/pools/src/lib.rs:801]: I think same here, `PoolChangesOf<T>` would be nice to have

**Review by branan — CHANGES_REQUESTED** [review]
Very small change requested for cleanup/readability. Otherwise this LGTM.

**Comment by lemunozm** [comment]
> Can anybody confirm that an extrinsic with an BoundedVec<T, S> will fail to be called when the input for the vector exceeds S?

Yes, it fails: https://paritytech.github.io/substrate/master/src/sp_core/bounded/bounded_vec.rs.html#81

**Comment by branan** [comment]
@mustermeiszer `BoundedVec` does check its bounds on decode - https://paritytech.github.io/substrate/master/src/sp_core/bounded/bounded_vec.rs.html#279-291

**Review by mustermeiszer — APPROVED** [review]
LGTM! Although, if you wanna adjust according to Branans proposal of creating a `type` for the changed structs that would be great.

File: pallets/pools/src/lib.rs
Comment (lemunozm) [reply pallets/pools/src/lib.rs:445]: Done, cleaner!

File: pallets/pools/src/lib.rs
Comment (lemunozm) [reply pallets/pools/src/lib.rs]: Done

**Review by branan — DISMISSED** [review]

**Comment by branan** [comment]
Removed my request for changes, so I _believe_ once this is rebased it can be approved by anyone to get it merged now

File: pallets/pools/src/benchmarking.rs
Comment (branan) [inline pallets/pools/src/benchmarking.rs:428]: You added a `ScheduledUpdateDetailsOf` type alias, you should use it here

**Review by branan — APPROVED** [review]
I turned off the auto-merge because I had a nitpick, but I'm otherwise happy with this :+1:

**Comment by branan** [comment]
Meh, changed my mind. This has been lingering long enough, and it's easy enough to fix that type the next time we touch this. As long as clippy isn't complaining about type complexity yet it's not the end of the world

**Comment by lemunozm** [comment]
I've seen late your last comment. Thanks for the review anyway!

**Pull request merged**
